### PR TITLE
Setup global error messages, added goroutine tests and code cleanup

### PR DESCRIPTION
**Pull request by nyghtowl**
Added global error messages and goroutine tests. Cleaned up and consolidated code, added and improved tests esp. for google groups.

### CODE REVIEW

File: 1-raw-data/utils/common_func.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		//} else {
+		err = fmt.Errorf("%w reading bodybytes failed: %v", httpStrRespErr, err)
+		return
+	}
+
+	responseString = string(bodyBytes)
+	return
+}
+
+type HttpDomeResponse func(string) (*goquery.Document, error)
```
Comment (juliaferraioli) [inline 1-raw-data/utils/common_func.go:62]: Please write a comment describing this type. Additionally, did you want `dome` or `dom`?

File: 1-raw-data/utils/fake_test_content.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/google/project-OCEAN/1-raw-data/gcs"
+)
+
+type FakeStorageConnection struct {
```
Comment (juliaferraioli) [inline 1-raw-data/utils/fake_test_content.go:26]: Please add a type description.

File: 1-raw-data/utils/fake_test_content.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	gcs.StorageConnection
+	ProjectID  string
+	BucketName string
+}
+
+func NewFakeStorageConnection(testPackage string) *FakeStorageConnection {
```
Comment (juliaferraioli) [inline 1-raw-data/utils/fake_test_content.go:32]: Please add a function comment.

File: 1-raw-data/utils/fake_test_content.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		return
+	} else if strings.Contains(content, "Susan") {
+		return fmt.Errorf("%w", storageErr)
+	} else if strings.Contains(content, "missing") {
+		return fmt.Errorf("Pipermail storage error")
+	}
+	return
+}
+
+func FakeHttpstringResponse(url string) (responseString string, err error) {
```
Comment (juliaferraioli) [inline 1-raw-data/utils/fake_test_content.go:54]: Please leave a function comment.

File: 1-raw-data/utils/fake_test_content.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	responseString = url
+	return
+}
+
+func FakeHttpDomResponse(url string) (dom *goquery.Document, err error) {
```
Comment (juliaferraioli) [inline 1-raw-data/utils/fake_test_content.go:59]: Please leave a function comment.

File: 1-raw-data/utils/fake_test_content.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+			<table>
+			<tr><td><a href="missing-chief.gz"</a></td></tr>
+			</table>
+			</html>
+    `
+	}
+	return goquery.NewDocumentFromReader(strings.NewReader(exDomResponse))
+}
+
+func FakeTopicIDToRawMsgUrlMap(org, groupName string, dom *goquery.Document) (rawMsgUrlMap map[string][]string, err error) {
```
Comment (juliaferraioli) [inline 1-raw-data/utils/fake_test_content.go:137]: Please leave a function comment.

**Review by juliaferraioli — CHANGES_REQUESTED** [review]
It's looking great! I'd just keep in mind the readability for someone who isn't you. Exported functions and types should _always_ have comments.

File: 1-raw-data/utils/common_func.go (hunk 1 above)
Comment (nyghtowl) [reply 1-raw-data/utils/common_func.go:62]: Good catch

**Comment by nyghtowl** [comment]
@teeler How does this look?

**Review by teeler — APPROVED** [review]
Only one tiny comment just to point it out - this looks really great though

File: 1-raw-data/gcs/gcs.go [hunk 7]
```diff
@@ -33,9 +33,17 @@ import (
 	"strings"
 )
 
+var (
+	httpStrRespErr     = fmt.Errorf("http string")
```
[the remark this answers is not in the document]
Comment (nyghtowl) [reply 1-raw-data/gcs/gcs.go:37]: Danka

**Pull request merged**